java/src/jmri/jmrix/sprog/SprogProgrammer.java

```diff
@@ -80,30 +83,67 @@ synchronized public void confirmCV(String CV, int val, jmri.ProgListener p) thro
      */
     @Override
     synchronized public void readCV(String CVname, jmri.ProgListener p) throws jmri.ProgrammerException {
+        readCVWithDefault(CVname, p, NO_DEFAULT_CV);
+    }
+    
+    /** 
+     * {@inheritDoc}
+     */
+    @Override
+    public void readCV(String CVname, jmri.ProgListener p, int startVal) throws jmri.ProgrammerException {
+        if (_sv != null) {
+            if (_sv.supportsCVHints()) {
+                // Connected hardware supports CV hint
+                log.debug("Hardware supports hints");
+                readCVWithDefault(CVname, p, startVal);
+            } else {
+                // Fallback to not using a hint
+                log.debug("Hardware does not support hints");
+                readCVWithDefault(CVname, p, NO_DEFAULT_CV);
+            }
+        } else {
+            // The SPROG version is not known yet so request the version for later
+            // and fall back to normal CV read this time
+            log.debug("SPROG version is unknown");
+            readCVWithDefault(CVname, p, NO_DEFAULT_CV);
+            _memo.getSprogVersionQuery().requestVersion(this);
+        }
+    }
+
+    /**
+     * Internal method to read a CV with a possible default value
+     * 
+     * @param CVname    Index of CV to read
+     * @param p         Programming listener
+     * @param startVal  CV default value, Use NO_DEFAULT_CV if no default available
+     * @throws jmri.ProgrammerException if programming operation fails
+     */
+    synchronized public void readCVWithDefault(String CVname, jmri.ProgListener p, int startVal) throws jmri.ProgrammerException {
         final int CV = Integer.parseInt(CVname);
         if (log.isDebugEnabled()) {
-            log.debug("readCV {} mode {} listens {}", CV, getMode(), p);
+            log.debug("readCV {} mode {} hint {} listens {}", CV, getMode(), startVal, p);
         }
         useProgrammer(p);
         _val = -1;
-        startProgramming(_val, CV);
+        startProgramming(_val, CV, startVal);
     }
 
     private jmri.ProgListener _usingProgrammer = null;
 
     /**
      * Send the command to start programming operation.
      * 
-     * @param val   Value to be written, or -1 for read
-     * @param CV    CV to read/write
+     * @param val       Value to be written, or -1 for read
+     * @param CV        CV to read/write
+     * @param startVal  Hint of what current CV value may be
```

Comment:
I suggest you change this to:
`@param startVal  Hint of what current CV value may be, or NO_DEFAULT_CV if no default.`